File: persistence-test/src/test/java/org/mariella/test/model/Folder.java
package org.mariella.test.model;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Inheritance;
import jakarta.persistence.InheritanceType;
import jakarta.persistence.Table;

@jakarta.persistence.Entity
@Table(name = "RESOURCE")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorValue("Folder")
public class Folder extends Resource {

}

File: persistence-kotlin/src/test/kotlin/org/mariella/persistence/kotlin/entities/Folder.kt
package org.mariella.persistence.kotlin.entities

import jakarta.persistence.*
import jakarta.persistence.Entity

@Entity
@Table(name = "folder")
@PrimaryKeyJoinColumn(name = "id", referencedColumnName = "id")
@Inheritance(strategy = InheritanceType.JOINED)
@DiscriminatorValue("FO")
class Folder : Resource() {
    override fun isContainer(): Boolean {
        return true
    }
}